src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
-
+Animal.alive = [];
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      if (animal.health <= 0) {
+        return;
+      }
+      animal.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter(({ health }) => health > 0);
```

Comment:
Do this only when the current animal has health <= 0.

Reply:
I do not get it, tests do not pass with this fix